opal/util/arch.h

```diff
@@ -202,7 +203,8 @@
 
 /* BYTE 2 */
 #define OPAL_ARCH_LONGISxx        0x0000c000  /* mask for sizeof long */
-#define OPAL_ARCH_LONGIS64        0x00001000
+#define OPAL_ARCH_LONGIS32        0x00004000
```

Comment:
As I mentionned above, I would prefer to leave the messy types to the OMPI layer, and keep the OPAL datatype simpler ([u]int[8,16,32,64,128]_t).